diff -p -up gfan0.4plus/polynomial.cpp.orig gfan0.4plus/polynomial.cpp
--- gfan0.4plus/polynomial.cpp.orig	2010-03-01 19:37:19.234485362 -0300
+++ gfan0.4plus/polynomial.cpp	2010-03-01 19:37:50.196484821 -0300
@@ -463,10 +463,8 @@ Polynomial Polynomial::homogenization(Po
   if(isMarked())
     {
 	  IntegerVector v=m;
-	  if(m.size()+1==newRing.getNumberOfVariables())
+	  if(m.size()==newRing.getNumberOfVariables())
 	  {
-	  v.grow(m.size()+1);
-	  v[m.size()]=degree-v.sum();
 	  ret.mark(Monomial(newRing,v));
 	  }
 	  //      assert(m.size()==newRing.getNumberOfVariables());
